central/sensor/service/pipeline/imageintegrations/pipeline.go

```diff
@@ -98,10 +90,20 @@ func (s *pipelineImpl) Reconcile(ctx context.Context, clusterID string, storeMap
 		if integration.GetClusterId() != clusterID {
 			continue
 		}
+
+		if !features.SourcedAutogeneratedIntegrations.Enabled() &&
+			env.AutogenerateGlobalPullSecRegistries.BooleanSetting() &&
```

Comment:
forgot to mention this before, but can you put these outside the for loop so we don't have to keep evaluating them every iteration? Maybe make it a global var so it can be used for other functions in this file